modules/packages/yum.in

```diff
@@ -123,6 +123,12 @@ def list_updates(online):
     # 0 means no updates
     # 1 means there was an error
     # 100 means that there are available updates
+    if process.returncode == 1 and not online:
+        # If we get an error when listing local updates, try again using the
+        # online method, so that the cache is generated
+        online_flag = []
+        process = subprocess_Popen([yum_cmd] + yum_options + online_flag + ["check-update"], stdout=subprocess.PIPE)
```

Comment:
You can just omit `online_flag` from the `Popen()` call instead of setting it to an empty list and then using it anyway.

Reply:
ACK. done and squashed.